odoorpc/rpc/jsonrpclib.py

```diff
@@ -49,12 +49,16 @@ def get_json_log_data(data):
     """Returns a new `data` dictionary with hidden params
     for log purpose.
     """
-    log_data = data
+    log_data = copy.deepcopy(data)
     for param in LOG_HIDDEN_JSON_PARAMS:
         if param in data['params']:
-            if log_data is data:
-                log_data = copy.deepcopy(data)
             log_data['params'][param] = "**********"
+
+    # The password is the 3rd element of the args array.
+    if 'args' in data['params']:
+        if 2 in data['params']['args']:
+            log_data['params']['args'][2] = "**********"
```

Comment:
I think, [reading this line](https://github.com/OCA/odoorpc/blob/c93ebd0221e1bf6a6ca380d614b07c59cf11732e/odoorpc/odoo.py#L349) `args` is a list, not a `dict`, can you confirm? 

```python
>>> test = ["a", "b", "c"]
>>> 2 in test
False
>>> test[2]
'c'
```

It could be nice to add a unit test of this use case ?

As this is a proxy, I guess there are other method going over this method that could be legitimate logged. I would suggest something likes


```suggestion
    # The password is the 3rd element of the args array.
    if 'args' in data['params'] and data['params'].get("method") == "login":
        if len(data['params']['args']) >= 3:
            log_data['params']['args'][2] = "**********"
```
you may check common service as well I suppose ?!

Reply:
I agree, method accepting passwords have to be checked 👍

Reply:
Details: the password is sent on every call sadly on `/jsonrpc` endpoint, both for the initial [`login` (from `common` service)](https://github.com/OCA/odoorpc/blob/c93ebd0221e1bf6a6ca380d614b07c59cf11732e/odoorpc/odoo.py#L343-L351) and then through [`execute` (from `object` service)](https://github.com/OCA/odoorpc/blob/c93ebd0221e1bf6a6ca380d614b07c59cf11732e/odoorpc/odoo.py#L361-L370).
So I would hide the password if the endpoint URL is `/jsonrpc` for `common` and `object` service. And regarding the [`db`](https://github.com/OCA/odoorpc/blob/master/odoorpc/db.py) service, we could keep things simple: hide all the `args` params (which contain super-admin password, database name etc).
There is also the `report` service but it's not working since Odoo 11.0, not sure we should try to handle it.

Reply:
This sounds very sensible. Personally I only ever use the web interface of Odoo and have no knowledge of the APIs other than tracking this issue internally at work once we discovered it.

I am totally fine with this PR being replaced by a more comprehensive fix.